File: rtsp2rtmp/comp_single/base/dto/po/base/_{snakeCase}__po.go
package po

import (
	"time"
)
// {{ rootInfo.displayName }}
type {{ rootInfo.pascalCaseName }}PO struct {
	// {{ rootInfo.pkAttributeInfo.displayName }}
	{{ rootInfo.pkAttributeInfo.pascalCaseName }} string  `json:"{{ rootInfo.pkAttributeInfo.camelCaseName }}"`
{%- for attributeInfo in rootInfo.baseAttributeInfoList %}
	// {{ attributeInfo.displayName }}{%- if attributeInfo.note %}{% else %}:{{ attributeInfo.note }}{%- endif %}
	{{ attributeInfo.pascalCaseName }} {{ attributeInfo.objectType }} `json:"{{ attributeInfo.camelCaseName }}"`
{%- endfor %}
{%- for attributeInfo in rootInfo.fkAttributeInfoList %}
	// {{ attributeInfo.displayName }}{%- if attributeInfo.note %}{% else %}:{{ attributeInfo.note }}{%- endif %}
	{{ attributeInfo.pascalCaseName }} string `json:"{{ attributeInfo.camelCaseName }}"`
{%- endfor %}
{%- for attributeInfo in rootInfo.upAttributeInfoList %}
	// {{ attributeInfo.displayName }}{%- if attributeInfo.note %}{% else %}:{{ attributeInfo.note }}{%- endif %}
	// {{ attributeInfo.pascalCaseName }} {{ attributeInfo.objectType }} `json:"{{ attributeInfo.camelCaseName }}"`
{%- endfor %}
{%- for attributeInfo in rootInfo.upSingleAttributeInfoList %}
	// {{ attributeInfo.displayName }}{%- if attributeInfo.note %}{% else %}:{{ attributeInfo.note }}{%- endif %}
	// {{ attributeInfo.pascalCaseName }} {{ attributeInfo.objectType }} `json:"{{ attributeInfo.camelCaseName }}"`
{%- endfor %}
{%- for attributeInfo in rootInfo.downAttributeInfoList %}
	// {{ attributeInfo.displayName }}{%- if attributeInfo.note %}{% else %}:{{ attributeInfo.note }}{%- endif %}
	// {{ attributeInfo.pascalCaseName }} []{{ attributeInfo.objectType }} `json:"{{ attributeInfo.camelCaseName }}"`
{%- endfor %}
{%- for attributeInfo in rootInfo.downSingleAttributeInfoList %}
	// {{ attributeInfo.displayName }}{%- if attributeInfo.note %}{% else %}:{{ attributeInfo.note }}{%- endif %}
	// {{ attributeInfo.pascalCaseName }} {{ attributeInfo.objectType }} `json:"{{ attributeInfo.camelCaseName }}"`
{%- endfor %}
}

File: tcdt_rust/dto_input_full/_{snakeCase}__po.rs
use serde::{Deserialize, Serialize};

{%- for entityInfo in rootInfo.entities %}
/// {{ entityInfo.displayName }}
#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct {{ entityInfo.pascalCaseName }}PO {
    #[serde(default)]
    pub action: i32,
    #[serde(default)]
    pub {{ entityInfo.pkAttributeInfo.snakeCaseName }}: String,
    {%- for attributeInfo in entityInfo.baseAttributeInfoList %}
        {%- if attributeInfo.attributeTypeCode is matching("xxxxxxx") %}
        {%- else %}
    /// {{ attributeInfo.displayName }}
    #[serde(default)]
    pub {{ attributeInfo.snakeCaseName }}: Option<{{ attributeInfo.objectType }}>,
        {%- endif %}
    {%- endfor %}
    {%- for attributeInfo in entityInfo.fkAttributeInfoList %}
    /// {{ attributeInfo.displayName }}
    #[serde(default)]
    pub {{ attributeInfo.snakeCaseName }}: Option<String>,
    {%- endfor %}
    {%- for attributeInfo in entityInfo.upAttributeInfoList %}
    /// {{ attributeInfo.displayName }}
    #[serde(default)]
    pub {{ attributeInfo.snakeCaseName }}: Option<{{ attributeInfo.objectType }}PO>,
    {%- endfor %}
    {%- for attributeInfo in entityInfo.upSingleAttributeInfoList %}
    /// {{ attributeInfo.displayName }}
    #[serde(default)]
    pub {{ attributeInfo.snakeCaseName }}: Option<{{ attributeInfo.objectType }}PO>,
    {%- endfor %}
    {%- for attributeInfo in entityInfo.downAttributeInfoList %}
    /// {{ attributeInfo.displayName }}
    #[serde(default)]
    pub {{ attributeInfo.snakeCaseName }}: Vec<{{ attributeInfo.objectType }}PO>,
    {%- endfor %}
    {%- for attributeInfo in entityInfo.downSingleAttributeInfoList %}
    /// {{ attributeInfo.displayName }}
    #[serde(default)]
    pub {{ attributeInfo.snakeCaseName }}: Option<{{ attributeInfo.objectType }}PO>,
    {%- endfor %}
}

{%- endfor %}
